src/sim_engine/pool.rs

```diff
@@ -59,10 +66,18 @@ impl Pool for SimPool {
         let pool_copy = SimPool {
             name: self.name.clone(),
             block_devs: self.block_devs.clone(),
+            filesystems: self.filesystems.clone(),
             raid_level: self.raid_level.clone(),
             online: true,
             checking: false,
         };
         Box::new(pool_copy)
     }
+    fn create_filesystem(&mut self,
+                         _filesystem_name: &str,
+                         _mount_point: &str,
+                         _size: &str)
```

Comment:
We should take the opportunity to define exactly what Sectors represent in that case and why it is suitable or adequate for this method. Right now, our design document does not use the word "sector" at all.

It looks like we can get rid of SumSectors pretty soon, based on progress of https://github.com/rust-lang/rust/issues/34529.

I am actually the world's leading expert in the representation of and computation with allocated address ranges for block storage, see: https://github.com/mulkieran/justbytes, and https://mojo.redhat.com/docs/DOC-1089158. (It's like being on the North Korean Olympic curling team). But, it would be a shame to waste my expertise in this area, if it would be useful.